packages/jest-core/src/runJest.ts

```diff
@@ -192,6 +192,7 @@ export default async function runJest({
     }),
   );
 
+  allTests = sequencer.shard?.(allTests, globalConfig.shard) ?? allTests;
```

Comment:
we should print a warning if the sequencer doesn't support the `shard` method and `globalConfig.shard` is set.

Or even throw

Reply:
I wondered that too - I erred towards retaining full backwards compatibility; users might have opted to implement the `TestSequencer` interface instead of inheriting the from the class. That being said happy to make this an exception.

Reply:
Yeah, and that sequencer is this not compatible with `shard` - I think it's fine to fail at that point. Either add support for `shard` or don't use the option - supplying it and it just not working feels like many bug reports waiting to happen 🙂